### PR TITLE
Elevators

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
It's a shuttle subtype that skips transit/ignition/ripples etc. Just simple up/down.

Not a very exciting PR

### CODE REVIEW

**Comment by RemieRichards** [comment]
does it still knock people over? it shouldn't.

**Comment by ma44** [comment]
Does it kill if someone is in the space on the other end?

**Comment by RemieRichards** [comment]
@ma44 yes.

**Comment by KorPhaeron** [comment]
@RemieRichards 

The knocking over occurs when it calls onshuttlemove on everything in the shuttle. Should I add an argument to onshuttlemove (ELEVATOR) and then check that in carbons onshuttlemove or can you think of a better way?

**Comment by RemieRichards** [comment]
@KorPhaeron an argument is fine.
Also, when someone is crushed by these elevators they get "hit by a hyperspace ripple" (visible message) still, should probably fix that.

File: code/modules/shuttle/on_move.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	return
 
 // Called when shuttle attempts to move an atom.
-/atom/movable/proc/onShuttleMove(turf/T1, rotation)
+/atom/movable/proc/onShuttleMove(turf/T1, rotation, force = FALSE)
```
Comment (RemieRichards) [inline code/modules/shuttle/on_move.dm]: bad arg name.

File: code/modules/shuttle/shuttle.dm [hunk 2]
```diff
@@ -517,7 +517,7 @@
 
 			//move mobile to new location
 			for(var/atom/movable/AM in T0)
-				if(AM.onShuttleMove(T1, rotation))
+				if(AM.onShuttleMove(T1, rotation, force))
```
Comment (RemieRichards) [inline code/modules/shuttle/shuttle.dm]: potential change in functionality, make this argument source its value from the docking port object, and set it to false on your elevators.
eg:
```
/obj/docking_port/elevator
    knockdown = FALSE
```

File: code/modules/shuttle/shuttle.dm [hunk 3]
```diff
@@ -578,10 +578,7 @@
 					if(M.pulledby)
 						M.pulledby.stop_pulling()
 					M.stop_pulling()
-					M.visible_message("<span class='warning'>[M] is hit by \
-							a hyperspace ripple!</span>",
-							"<span class='userdanger'>You feel an immense \
-							crushing pressure as the space around you ripples.</span>")
+					M.visible_message("<span class='warning'>[src] slames into [M]!</span>"
```
Comment (RemieRichards) [inline code/modules/shuttle/shuttle.dm]: slames

**Comment by GunHog** [comment]
On the actual Planetstation, how hard would this be to do with the Aux Base?

**Comment by Dorsisdwarf** [comment]
Can we get a rattling chain lift sound for the transition

**Comment by Jalleo** [comment]
Ask Cuboos for that.

**Comment by ExcessiveUseOfCobblestone** [comment]
https://www.youtube.com/watch?v=RCSbmSmyAcY

for transition.

**Comment by coiax** [comment]
Did you mean to change regular shuttles to not hit people with hyperspace ripples anymore?

**Comment by KorPhaeron** [comment]
Was trying to think of a generic message that would work for both

**Pull request merged**